test cases/common/202 gcc func attributes/meson.build

```diff
@@ -0,0 +1,105 @@
+# Copyright © 2017-2018 Intel Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+project('gcc func attributes', ['c', 'cpp'])
+
+# For msvc these will fail because msvc doesn't support __attribute__, for
+# Clang and GCC, they should pass.
+c = meson.get_compiler('c')
+cpp = meson.get_compiler('cpp')
+
+# If the compiler isn't GCC, Clang, or MSVC we don't know what this will
+# return, so we don't run
+if not ['gcc', 'clang', 'msvc'].contains(c.get_id())
+  error('MESON_SKIP_TEST: Compiler __attribute__ support unknown.')
```

Comment:
ICC also supports gnu-style function attributes, but not all of them.

Reply:
i.e., we'll want to not skip the test, but instead run it and get someone who cares about ICC to fix it for us. You don't need to test ICC by hand and make the test pass.